Sources/Confidence/ConfidenceScreenTracker.swift

```diff
@@ -0,0 +1,102 @@
+#if os(iOS) || os(tvOS) || os(visionOS) || targetEnvironment(macCatalyst)
+import Foundation
+import UIKit
+import Combine
+
+public class ConfidenceScreenTracker: ConfidenceEventProducer {
```

Comment:
I am not sure how to test this. I think it's a nice to have, but not required in this first PR. Since it's optional for users, I am also ok with merging it anyway

Reply:
we need another demo app that uses UIKit instead of SwiftUI, there we can test this with multiple viewControllers properly. I'd say let's keep it but only use it when that demo app is created.

Reply:
I am not sure if we should keep this without having any sort of unit test and or manual test capabilities